pact/sale-contracts/dutch-auction/dutch-auction.pact

```diff
@@ -0,0 +1,210 @@
+(namespace (read-msg 'ns))
+
+(module dutch-auction GOVERNANCE
+  @doc "Dutch auction contract"
+
+  (defconst ADMIN-KS:string "marmalade-sale.marmalade-contract-admin")
+
+  (defcap GOVERNANCE ()
+    (enforce-guard ADMIN-KS))
+
+  (use marmalade-v2.policy-manager)
+  (use marmalade-v2.policy-manager [BUYER-FUNGIBLE-ACCOUNT-MSG-KEY])
+  (implements marmalade-v2.sale-v2)
+
+  (defschema auctions-schema
+    token-id:string
+    start-date:integer
+    end-date:integer
+    start-price:decimal
+    reserve-price:decimal
+    sell-price:decimal
+    buyer:string
+    buyer-guard:guard
+  )
+
+  (deftable auctions:{auctions-schema})
+
+  (defcap AUCTION_CREATED:bool
+    ( sale-id:string
+      token-id:string
+    )
+    @event
+    true
+  )
+
+  (defcap MANAGE_AUCTION:bool (sale-id:string token-id:string)
+    (let* (
+      (quote-info:object{quote-schema} (get-quote-info sale-id))
+      (seller:string (at 'seller quote-info)))
+      (enforce-guard (marmalade-v2.ledger.account-guard token-id seller))
+    )
+  )
+
+  (defcap PRICE_ACCEPTED:bool
+    ( sale-id:string
+      bidder:string
+      bidder-guard:guard
+      price:decimal
+      token-id:string
+    )
+    @event
+    true
+  )
+
+  (defcap DUMMY:bool () true)
+  (defconst DUMMY_GUARD:guard (create-capability-guard (DUMMY)))
+
+  (defun enforce-fungible-transfer:bool (sale-id:string)
```

Comment:
Doesn't seem it's needed here, because dutch auction does not involve an escrow account

Reply:
The dummy guard is used to insert into the table upon auction creation, as a placeholder:

`,"reserve-price": reserve-price
        ,"sell-price": 0.0
        ,"buyer": ""
        ,"buyer-guard": DUMMY_GUARD
      })`